### ISSUE TITLE
vendor/github.com/openshift/client-go/build/clientset/versioned/typed/build/v1/build_client.go:61:32: undefined: serializer.DirectCodecFactory

### ISSUE THREAD

**Issue by shariefmohammed** [openshift/client-go#134]
I used  the code as is and try to build the code . 
**go build -o bin/getnode ao.dcn/simple-golang-test/pkg**
I get below error 

vendor/github.com/openshift/client-go/build/clientset/versioned/typed/build/v1/build_client.go:61:32: undefined: serializer.DirectCodecFactory

**Comment by porgull** [comment]
I had the same error. It's because it will automatically download the latest version of client-go/apimachinery/etc. when building with go. So, to fix, you have to hard-code the correct version of the k8s api that you want, and ensure that it's consistent across the different APIs you import. For example, with go modules:

```
module example.com/example/example

go 1.13

require (
...
	k8s.io/api v0.15.9
	k8s.io/apimachinery v0.15.9
	k8s.io/cli-runtime v0.15.9 // indirect
	k8s.io/client-go v0.15.9
	k8s.io/component-base v0.15.9 // indirect
	k8s.io/kubectl v0.15.9 // indirect
	k8s.io/metrics v0.15.9 // indirect
...
)

```

If you make all of the versions consistent, then there shouldn't be any import issues.

**Comment by jkandasa** [comment]
@shariefmohammed I too face this issue. In this repository the last released tag is v3.9.0.
By default `go.mod` is taking `github.com/openshift/client-go v3.9.0+incompatible` as dependent.
We should take it from `master` branch for the most recent update.

delete all the required libraries on your `go.mod` file and run the following command on your project,
```
go get github.com/openshift/client-go@master
```

When I write this comment, My `go.mod` looks like this,
```
require (
	github.com/openshift/api v0.0.0-20200116145750-0e2ff1e215dd
	github.com/openshift/client-go v0.0.0-20200116152001-92a2713fa240
	k8s.io/apimachinery v0.17.1
	k8s.io/client-go v0.17.1
)
```

And the issue is gone :)

Source: https://github.com/golang/go/wiki/Modules#how-to-upgrade-and-downgrade-dependencies
```
Using a branch name such as go get foo@master (foo@tip with mercurial) is one way to obtain the latest commit regardless of whether or not it has a semver tag.
```
